tests/unit/test_network_connections.py

```
@@ -5592,11 +5592,75 @@ def test_invalid_pkey_values(self):
+                    "perm-address": None,
+                    "bond-port-perm-hwaddr": None,
+                }
+            }
+
+        # Use Mock(side_effect=...) so Py2.7 does not treat the replacement as an
+        # unbound method (plain functions patched onto @staticmethod fail there).
+        fetch_mock = mock.Mock(side_effect=unstable_fetch)
+        with mock.patch.object(SysUtil, "_link_infos_fetch", fetch_mock):
+            try:
+                SysUtil.link_infos(refresh=True)
+            except Exception as exc:
```

Ruff (0.15.13)

[warning] Do not catch blind exception: `Exception`

(BLE001)